### PR TITLE
Optimization

### PR DESCRIPTION
**Pull request by mareklibra**
- Shared vm icons are loaded just once
- the vms reducer supports bulk updates (less actions)

### CODE REVIEW

File: src/components/VmDetail/index.js [hunk 1]
```diff
@@ -71,6 +73,7 @@ class VmDetail extends Component {
 }
 VmDetail.propTypes = {
   vm: React.PropTypes.object,
```
Comment (matobet) [inline src/components/VmDetail/index.js:76]: Now thinking of this: isn't the `vm` property also required? :)

File: src/reducers/icons.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * The Icons reducer
+ *
+ * @param state
+ * @param action
+ * @returns {*}
+ */
+function icons (state, action) {
+  state = state || Immutable.fromJS({})
```
Comment (matobet) [inline src/reducers/icons.js:11]: please prefer `Immutable.Map` to `Immutable.fromJS({})`. Overall I would suggest restructuring this code a bit:

``` javascript
import { Map } From 'immutable'

function icons (state = Map(), action) {
    // ... rest of the code
}
```

File: src/reducers/vms.js [hunk 3]
```diff
@@ -59,10 +52,8 @@ function vms (state, action) {
   logDebug(`The 'vms' reducer action=${JSON.stringify(hidePassword({ action }))}`)
 
   switch (action.type) {
-    case 'UPDATE_VM':
+    case 'UPDATE_VMS':
```
Comment (matobet) [inline src/reducers/vms.js:55]: where is the respective actionCreator change corresponding to this `UPDATE_VM -> UPDATE_VMS` change?

File: src/reducers/icons.js (hunk 2 above)
Comment (mareklibra) [reply src/reducers/icons.js:11]: ok

File: src/reducers/vms.js (hunk 3 above)
Comment (mareklibra) [reply src/reducers/vms.js:55]: Good point, accidentally moved to UserPortal.
Moving back, to be collocated with the reducer

**Pull request closed without merging**